services/can_logger/can_logger.py

```diff
@@ -8,51 +7,124 @@
 import sys
 import collections
 import multiprocessing as mp
+import logging
 from datetime import datetime
 from psycopg2 import OperationalError
 from psycopg2.extras import execute_values
 
-
 def write_to_csv(wr_buff, can_bus):
-
     # Write to CSV from write buffer. First item is timestamp, second is
-    # can_id and third is can_data
-
-    with open(f'/data/log/{can_bus}.csv', mode='a') as logfd:
-
-        logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
-                            quoting=csv.QUOTE_MINIMAL)
-
-        logcsv.writerows(wr_buff)
+    # can_bus, third is can_id, and last is can_data
+    # Generate timestamp to rotate the log
+    global log_path
+    #timestamp = rotate_log()
+    while(not log_path):
+        try:
+            with open(f'/data/log/can/{can_bus}.csv', mode='a') as logfd:
+                logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
+                                    quoting=csv.QUOTE_MINIMAL)
+                logcsv.writerows(wr_buff)
+                log_path = True
+        except(FileNotFoundError):
+            path = '/data/log/can'
+            os.mkdir(path)
+            print(f'Logging path set to {path}')
 
     print(f'{can_bus}: Wrote successfully to CSV')
 
+def rotate_log(can_bus, interval, period):
```

Comment:
Somewhat of a nit: wouldn't accepting `interval` as seconds directly remove complexity and increase capability?

Reply:
It is true. However, the way I thought the log rotation depends on `time.strftime(time.time())`, which depending on the formatting options (set by the flag -INTERVAL) will return the numeric value of the current seconds, minutes and, days. Then we use this output to compare against a value previously stored in order to keep track of interval changes. This is likely to change in the future for the NATS-based logger.